src/app/components/basket/basket.component.spec.ts

```diff
@@ -13,11 +15,101 @@ describe('BasketComponent', () => {
       .compileComponents();
 
     fixture = TestBed.createComponent(BasketComponent);
+    fixture.componentRef.setInput('items', []);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should add an item to the basket', () => {
+    const item: Item = {
+      id: 1,
+      name: 'item',
+    };
+    fixture.componentRef.setInput('items', [item]);
+
+    const newItems = component.items();
+    expect(newItems).toEqual([item]);
+  });
+
+  it('should contain item id and name', () => {
+    const item: Item = {
+      id: 1,
+      name: 'fancyItemName',
+    };
+    fixture.componentRef.setInput('items', [item]);
+    fixture.detectChanges();
+
+    const itemsTable = fixture.nativeElement.querySelector('table');
+
+    expect(itemsTable.textContent).toContain('1');
+    expect(itemsTable.textContent).toContain('fancyItemName');
+  });
+
+  it('should not have a remove button if not an admin', () => {
+    const item: Item = {
+      id: 1,
+      name: 'item',
+    };
+    fixture.componentRef.setInput('items', [item]);
+    fixture.detectChanges();
+
+    const removeButton = fixture.nativeElement.querySelector('button');
+
+    expect(removeButton).toBeNull();
+  })
+
+  it('should have a remove button if an admin', () => {
+    const item: Item = {
+      id: 1,
+      name: 'item',
+    };
+
+    fixture.componentRef.setInput('items', [item]);
+    fixture.componentRef.setInput('isAdmin', true);
+    fixture.detectChanges();
+
+    const removeButton = fixture.nativeElement.querySelector('button');
+
+    expect(removeButton).not.toBeNull();
+  });
+
+  it('should call the item to remove', () => {
+    const item: Item = {
+      id: 1,
+      name: 'item',
+    };
+    fixture.componentRef.setInput('items', [item]);
+    fixture.componentRef.setInput('isAdmin', true);
+    spyOn(component, 'removeItemById');
```

Comment:
Good job with the `spyOn`, that's exactly what we want to do here.
Furthermore, we can also check if an `output` is emitted using the `component.onItemRemoveById` method. 
Also, I think `on<EVENT>` is discouraged by Angular for custom output emitters as Angular itself adds has `onCancel` etc for its internal events.

Reply:
Okay, I see that you have another test for it. We can combine those two in one test.